Documentation/guides/profiling.md

```diff
@@ -0,0 +1,244 @@
+# Profiling Startup
+
+There are currently a few different ways at getting timing information
+from Xamarin.Android & Mono. This document aims to demystify the
+command-line commands involved in retrieving this data from Android
+devices.
+
+This document explains many uses of `adb` at the command-line. On
+Windows, you will need either `adb.exe` in your `%PATH%` or a
+[powershell alias][set_alias] for `adb`.
+
+[set_alias]: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/set-alias?view=powershell-6
+
+## Overall Startup
+
+The easiest way to time the complete time of startup is to launch the
+app by tapping the icon, and run:
+
+    # Windows / Powershell
+    > adb logcat -d | Select-String Displayed
+
+    # Mac / Unix
+    > adb logcat -d | grep Displayed
+
+_NOTE: going forward, I will just use `grep` in examples. On Windows, keep in mind to use `Select-String` if needed._
+
+Both platforms will output something such as:
+
+    09-04 14:33:32.466  1169  1192 I ActivityManager: Displayed com.xamarin.android.helloworld/example.MainActivity: +1s3ms
+
+The `ActivityManager` system process logs how long it takes for any
+Android activity to start. This is the best measure of overall time
+taken: reflecting what a user experiences when launching the app.
+
+You can also run `adb logcat -c` to clear the log at any point.
+
+## Profiling the Runtime
+
+Before launching the app, run the `adb` command:
+
+    > adb shell setprop debug.mono.log default,timing
```

Comment:
Here it is usually more suitable to use `adb shell setprop debug.mono.log default,timing=bare` to avoid writing the JIT information, which affects the timing.